### PR TITLE
Feature add BE reviews

### LINKED ISSUE #11: Bonus BE: Reviews
Create a reviews model
- can contain multiple reviews
- each review is 1-5 and an optional text
Implement
- `GET` all reviews for user
- `POST` make a "create review" route for testing reasons
Add the reviews to a user in whatever way you see fit

### PR DESCRIPTION
**Pull request by navjot-chahal**
### What this PR does (required):
- BE setup for posting and retrieving reviews.
- Setup `GET` `/reviews/:profileID:` and `POST` `/reviews/` routes .
- Only allow posting reviews when the `reviewer` and `reviewed` are part of initial request and have opposite roles.
- Only allow review when the status of the corresponding request is `PAID`.

### Screenshots / Videos:
- Postman manual testing:

GET `/reviews/:profileId`

GET Reviews

POST `/reviews/`

POST Review

### Any information needed to test this feature:
- Need a pre-populated DB with a few `User`, `Profile`, and `Request` documents to test this.

closes #11

### CODE REVIEW

File: server/controllers/reviews.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+const Profile = require("../models/Profile");
+const Request = require("../models/Request");
+const Review = require("../models/Review");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /reviews/
+// @desc Post new review for a profile
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/reviews.js:8]: Good documentation

File: server/controllers/reviews.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+exports.postReview = asyncHandler(async (req, res, next) => {
+  const { reviewedProfileId, requestId, starRating, text } = req.body;
```
Comment (sundayezeilo) [inline server/controllers/reviews.js:10]: Validate request body and return a `400` if any of these is not provided

File: server/controllers/reviews.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    starRating,
+    text,
+  });
+
+  return res.status(201).json({ success: { review: newReview } });
+});
+
+// @route GET /reviews/:profileId
+// @desc Get existing reviews for a profileId
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/reviews.js:62]: :+1:

File: server/models/Review.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      ref: "Profile",
+    },
+    reviewedProfileId: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "Profile",
+    },
+    requestId: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "Request",
+    },
```
Comment (sundayezeilo) [inline server/models/Review.js]: Shouldn't any of these be `required`?

**Review by sundayezeilo — CHANGES_REQUESTED** [review]
Good work with this one. Just a few minor comments left for you to take a look at. Also, it is required to also link your ticket in all your PRs. Take a look at this [guide](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) on how to properly link an issue in your PR.
Happy coding!

File: server/controllers/reviews.js (hunk 2 above)
Comment (navjot-chahal) [reply server/controllers/reviews.js:10]: @ezeilo-su All of these fields are getting validated in a separate `validatePostReview` function with the express-validator.

File: server/models/Review.js (hunk 4 above)
Comment (navjot-chahal) [reply server/models/Review.js]: Yes, I will add these in the following commit!

**Comment by navjot-chahal** [comment]
@ezeilo-su I have added the "required tag fix" with the `Hotfix - 1` commit. Seemingly the `closes #11` closing statement did not link the issue automatically for some reason. 🤨

**Review by rajivtitus — APPROVED** [review]
LGTM! Great work with this 👍🏽

**Pull request merged**